lib/Controller/SocialApiController.php

```diff
@@ -0,0 +1,300 @@
+
+	//** @var IInitialStateService */
+	// private  $initialStateService;
+
+	/** @var IFactory */
+	private  $languageFactory;
+	/** @var IManager */
+	private  $manager;
+	/** @var IConfig */
+	private  $config;
+
+	/**
+	 * This constant stores the supported social networks
+	 * It is an ordered list, so that first listed items will be checked first
+	 * Each item stores the avatar-url-formula as recipe, a cleanup parameter to
+	 * extract the profile-id from the users entry, and possible filters to check
+	 * validity
+	 * 
+	 * @const {array} SOCIAL_CONNECTORS dictionary of supported social networks
+	 */
+	const SOCIAL_CONNECTORS = [
+		'facebook' 	=> [
+			'recipe' 	=> 'https://graph.facebook.com/{socialId}/picture?width=720',
+			'cleanups' 	=> ['basename'],
+			'checks'	=> [],
+		],
+		'tumblr' 	=> [
+			'recipe' 	=> 'https://api.tumblr.com/v2/blog/{socialId}/avatar/512',
+			'cleanups' 	=> ['filter'],
+			'filter' 	=> ['regex' => '/(?:http[s]*\:\/\/)*(.*?)\.(?=[^\/]*\..{2,5})/i', 'group' => 1], // "subdomain"
+			'checks'	=> [],
+		],
+		/* untrusted
+		'instagram' 	=> [
+			'recipe' 	=> 'http://avatars.io/instagram/{socialId}',
+			'cleanups' 	=> ['basename'],
+			'checks'	=> []
+		],
+		'twitter' 	=> [
+			'recipe' 	=> 'http://avatars.io/twitter/{socialId}',
+			'cleanups' 	=> ['basename'],
+			'checks'	=> []
+		],
+		*/
+	];
+
+	public function __construct(string $AppName,
+								IRequest $request,
+								IManager $manager,
+								IConfig $config,
+								// IInitialStateService $initialStateService,
+								IFactory $languageFactory) {
+		parent::__construct($AppName, $request);
+
+		$this->appName = $AppName;
+		// $this->initialStateService = $initialStateService;
+		$this->languageFactory = $languageFactory;
+		$this->manager = $manager;
+		$this->config = $config;
+
+	}
+
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * returns an array of supported social networks
+	 *
+	 * @param {String} type the kind of information interested in -- provision
+	 * @returns {array} an array of supported social networks
+	 */
+	public function getSupportedNetworks(string $type) : array {
+
+		$supported = array();
+		$supported['avatar'] = array();
+
+		foreach(array_keys(self::SOCIAL_CONNECTORS) as $network) {
+			array_push($supported['avatar'], $network);
+		}
+
+		if (strcmp($type, 'all') === 0) {
+			// return array of arrays
+			return $supported;
+		}
+		if (array_key_exists($type, $supported)) {
+			return $supported[$type];
+		}
+		// unknown type
+		return array();
+	}
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * generate download url for a social entry (based on type of data requested)
+	 *
+	 * @param {array} socialentries the network and id from the social profiles
+	 * @param {array} network the choice which network to use or 'first' to use any
+	 * @returns {String} the url to the requested information or null in case of errors
+	 */
+	protected function getSocialConnector(array $socialentries, string $network) : ?string {
```

Comment:
I find this super confusing. :thinking: 

Why are we providing an array as socialentries ?
Are we checking multiple url at the same time?

Should we use a more appropriate function name?

I'm also really not fond of having everything declared here. The more I think of it, the more I think we shuold split each SocialDataProvider into its own dedicated file. 
That would be much cleaner. I'm going to give it a few thoughts and get back here :)

Reply:
well, originally the array of social profiles was just passed to this function and the first supported (according to the set order) profile taken. For me, it also felt like the dedicated selection of social profiles (added only in the last commit) made it way more complex. Maybe we should set the dedicated selection aside until we have the favorite-option?

Reply:
Removed the type attribute for less complexity, made some cleanups and renamed the variables. Hope that helped.